Assets/MRTK/Providers/LeapMotion/Editor/ConfigurationChecker/LeapMotionConfigurationChecker.cs

```diff
@@ -0,0 +1,263 @@
+                    if (File.Exists(fullPath) && fullPath.Contains(".cs"))
+                    {
+                        // Delete the test files
+                        FileUtil.DeleteFileOrDirectory(fullPath);
+
+                        // Also delete the meta files
+                        FileUtil.DeleteFileOrDirectory(fullPath + ".meta");
+                    }
+
+                    if (Directory.Exists(fullPath) && !fullPath.Contains(".cs"))
+                    {
+                        // Delete the test directories
+                        FileUtil.DeleteFileOrDirectory(fullPath);
+
+                        // Delete the test directories meta files
+                        FileUtil.DeleteFileOrDirectory(fullPath.TrimEnd('/') + ".meta");
+                    }
+                }
+
+                AssetDatabase.Refresh();
+            }
+        }
+
+        /// <summary>
+        /// Adds an asmdef at the root of the LeapMotion Core Assets once they are imported into the project and adds the newly created LeapMotion.asmdef
+        /// as a reference for the existing leap data provider asmdef.
+        /// </summary>
+        private static void AddAndUpdateAsmDefs()
+        {
+            // Find where the leap motion core assets are relative to the assets directory
+            string pathDifference = GetPathDifference();
+
+            string leapCoreAsmDefPath = Path.Combine(Application.dataPath, pathDifference, "LeapMotion", "LeapMotion.asmdef");
+
+            // If the asmdef has already been created then do not create another one
+            if (!File.Exists(leapCoreAsmDefPath))
+            {
+                // Create the asmdef that will be placed in the Leap Core Assets when they are imported
+                // A new asmdef needs to be created in order to reference it in the MRTK/Providers/LeapMotion/Microsoft.MixedReality.Toolkit.Providers.LeapMotion.asmdef file
+                AssemblyDefinition leapAsmDef = new AssemblyDefinition
+                {
+                    Name = "LeapMotion",
+                    AllowUnsafeCode = true,
+                    References = new string[] { },
+                    IncludePlatforms = new string[] { "Editor", "WindowsStandalone32", "WindowsStandalone64"}
+                };
+
+                leapAsmDef.Save(leapCoreAsmDefPath);
+
+                // Get the MRTK/Providers/LeapMotion/Microsoft.MixedReality.Toolkit.Providers.LeapMotion.asmdef
+                FileInfo[] leapDataProviderAsmDefFile = FileUtilities.FindFilesInAssets("Microsoft.MixedReality.Toolkit.Providers.LeapMotion.asmdef");
+
+                // Add the newly created LeapMotion.asmdef to the references of the leap data provider asmdef
+                AssemblyDefinition leapDataProviderAsmDef = AssemblyDefinition.Load(leapDataProviderAsmDefFile[0].FullName);
+
+                leapDataProviderAsmDef.References = new string[]
+                { "Microsoft.MixedReality.Toolkit",
```

Comment:
Another comment around safety of overwriting what's there - if we ever change the content in Microsoft.MixedReality.Toolkit.Providers.LeapMotion.asmdef, this code here will probably blow it away - as a result it will need to be updated in two locations.

We'll either need some code here that handles it, or something that will error out/break in the CI pipeline to catch this.

Reply:
In other config checkers, we explictly append, and only when it doesn't already exist:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/3f9d102e2f43a7acc2f9f81edae32f84acc9bf2a/Assets/MRTK/Providers/XRSDK/Editor/XRSDKConfigurationChecker.cs#L81-L92

We could do that here, since I think the only thing we need to add is the `LeapMotion` reference?

It also gets removed in the case where the ref doesn't exist (but I think recent versions of 2018.4 and 2019 should handle this case without error):

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/3f9d102e2f43a7acc2f9f81edae32f84acc9bf2a/Assets/MRTK/Providers/XRSDK/Editor/XRSDKConfigurationChecker.cs#L107-L118

Reply:
Ah I see. Kurtis's suggestion has been added.